engine/src/main/java/org/terasology/rendering/dag/nodes/AlphaRejectBlocksNode.java

```diff
@@ -45,34 +50,57 @@
  * In alpha-blending the color of a semi-transparent fragment is combined with
  * the color stored in the frame buffer and the resulting color overwrites the previously stored one.
  */
-public class AlphaRejectBlocksNode extends WireframeCapableNode {
+public class AlphaRejectBlocksNode extends AbstractNode implements WireframeCapable {
 
     private static final ResourceUrn CHUNK_SHADER = new ResourceUrn("engine:prog.chunk");
 
     @In
     private ComponentSystemManager componentSystemManager;
 
     @In
+    private Config config;
+
+    @In
     private WorldRenderer worldRenderer;
 
     @In
     private RenderQueuesHelper renderQueues;
 
     private Camera playerCamera;
     private Material chunkShader;
+    private SetWireframe wireframeStateChange;
+    private RenderingDebugConfig renderingDebugConfig;
 
     /**
      * Initialises the node. -Must- be called once after instantiation.
      */
     @Override
     public void initialise() {
-        super.initialise();
         playerCamera = worldRenderer.getActiveCamera();
+
+        wireframeStateChange = new SetWireframe(true);
+        renderingDebugConfig = config.getRendering().getDebug();
+        new WireframeTrigger(renderingDebugConfig, this);
+
         addDesiredStateChange(new SetViewportToSizeOf(READ_ONLY_GBUFFER));
         addDesiredStateChange(new EnableMaterial(CHUNK_SHADER.toString()));
         chunkShader = getMaterial(CHUNK_SHADER);
     }
 
+    public void enableWireframe() {
+        if (!getDesiredStateChanges().contains(wireframeStateChange)) {
+            addDesiredStateChange(wireframeStateChange);
+            refreshTaskList();
+        }
+    }
+
+    public void disableWireframe() {
+        if (getDesiredStateChanges().contains(wireframeStateChange)) {
+            removeDesiredStateChange(wireframeStateChange);
+            refreshTaskList();
+        }
+    }
```

Comment:
I will take one StrawberryEatingCapableNode please. For ... personal reasons. 

:strawberry: :strawberry: :strawberry: 

At first I was thinking about suggesting default methods in the interfaces but needing to include `wireframeStateChange` as a variable would get in the way. Unless ... you used 

`removeDesiredStateChange(getWireframeStateChange());`

But then you'd still need to implement the getter anyway ... nevermind :(